### PR TITLE
test: add tests on mediaStreamHandler

### PR DESCRIPTION
**Pull request by atomrc**

### CODE REVIEW

File: test/unit_tests/media/MediaStreamHandlerSpec.js [hunk 1]
```diff
@@ -35,6 +35,81 @@ describe('z.media.MediaStreamHandler', () => {
...
+  describe('addRemoteMediaStream', () => {
+    it('throws an error if stream type is not recognized', () => {
+      const streamHandler = TestFactory.media_repository.stream_handler;
+
+      const newMediaStream = {type: 'random'};
+
+      try {
+        streamHandler.addRemoteMediaStream(newMediaStream);
+      } catch (error) {
+        expect(error instanceof z.media.MediaError).toEqual(true);
```
Comment (ffflorian) [inline test/unit_tests/media/MediaStreamHandlerSpec.js]:
```js
.toBe(true)
```

File: test/unit_tests/media/MediaStreamHandlerSpec.js [hunk 2]
```diff
@@ -35,6 +35,81 @@ describe('z.media.MediaStreamHandler', () => {
...
+        [recognizedStreams[0], recognizedStreams[1]],
+        [recognizedStreams[0], recognizedStreams[1], recognizedStreams[2]],
+      ];
+
+      const subscription = streamHandler.remoteMediaStreamInfo.subscribe(streams => {
+        expect(streams).toEqual(expectedStreams.shift());
+      });
+
+      recognizedStreams.forEach(stream => {
+        streamHandler.addRemoteMediaStream(stream);
```
Comment (bennycode) [inline test/unit_tests/media/MediaStreamHandlerSpec.js]: Can be done in 1 line. I remember we had a linting rule for that? @lipis

File: test/unit_tests/media/MediaStreamHandlerSpec.js [hunk 3]
```diff
@@ -35,6 +35,81 @@ describe('z.media.MediaStreamHandler', () => {
...
+        audioObservable.subscribe(audioStreams => {
+          expect(audioStreams).toEqual(expectedAudioStreams.shift());
+        })
+      );
+
+      subscriptions.push(
+        videoObservable.subscribe(videoStreams => {
+          expect(videoStreams).toEqual(expectedVideoStreams.shift());
+        })
+      );
```
Comment (ffflorian) [inline test/unit_tests/media/MediaStreamHandlerSpec.js]:
```js
const subscriptions = [
  audioObservable.subscribe(audioStreams => {
    expect(audioStreams).toEqual(expectedAudioStreams.shift());
  }),

  videoObservable.subscribe(videoStreams => {
    expect(videoStreams).toEqual(expectedVideoStreams.shift());
  })
];
```

File: test/unit_tests/media/MediaStreamHandlerSpec.js [hunk 4]
```diff
@@ -35,6 +35,81 @@ describe('z.media.MediaStreamHandler', () => {
...
+
+      [audioStream, videoStream, audioVideoStream].forEach(stream => {
+        streamHandler.addRemoteMediaStream(stream);
+      });
+
+      subscriptions.forEach(subscption => subscption.dispose());
```
Comment (ffflorian) [inline test/unit_tests/media/MediaStreamHandlerSpec.js]:
```js
subscriptions.forEach(subscription => subscription.dispose());
```

**Pull request merged**